src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java

```diff
@@ -213,24 +215,50 @@ public JurisdictionDefinition getJurisdictionFromDefinitionStore(String jurisdic
         return jurisdictionDefinitions.get(0);
     }
 
+    @Override
+    public List<String> getCaseTypesIDsByJurisdictions(List<String> jurisdictionIds) {
+
+        List<JurisdictionDefinition> jurisdictionDefinitions = getJurisdictionsFromDefinitionStore(jurisdictionIds);
+        if (jurisdictionDefinitions.isEmpty()) {
+            LOG.warn("Definitions not found for requested jurisdictions {}", jurisdictionIds);
+            return Collections.emptyList();
+        }
+        List<String> caseTypes = getCaseTypeIdFromJurisdictionDefinition(jurisdictionDefinitions);
+        return caseTypes;
+    }
+
+    @Override
+    public List<String> getAllCaseTypesIDs() {
+        List<JurisdictionDefinition> jurisdictionDefinitions = getJurisdictionsFromDefinitionStore(Collections.emptyList());
+        return getCaseTypeIdFromJurisdictionDefinition(jurisdictionDefinitions);
+    }
+
+    private List<String> getCaseTypeIdFromJurisdictionDefinition(List<JurisdictionDefinition> jurisdictionDefinitions) {
+        return jurisdictionDefinitions.stream().flatMap(
+            jurisdictionDefinition -> jurisdictionDefinition.getCaseTypeDefinitions().stream().map(
+                caseTypeDefinition -> caseTypeDefinition.getId()
```

Comment:
Minor: We could add a `getAllCaseTypes()` into `JurisdictionDefinition`

Reply:
done